tuist/tuist
# Support generating info.plist for Watch Apps & Extensions

Part of: https://github.com/tuist/tuist/issues/628

### Short description 📝

Add support for generating the Info.plist file for watchOS targets

### Solution 📦

Extend the Info plist content provider to support watchOS targets taking into account that:

- Watch apps reference their host applications bundle identifier in the info plist `WKCompanionAppBundleIdentifier` key

- Watch app extensions reference their host watch apps bundle identifier in the info plist `NSExtension.NSExtensionAttributes.WKAppBundleIdentifier`

### Implementation 👩‍💻👨‍💻

- [x] Add watchOS App / Extension defaults to the info plist content provider
- [x] Lookup host applications bundle identifiers

### Test Plan 🛠

- run `tuist generate` within `fixtures/ios_app_with_watchapp2`
- Verify the info.plist files generated in `Derrived/InfoPlists` matche the ones created by Xcode
  (They were previously checked in under `Support`)
